File: C++/A1/052.cpp
#include <iostream>
#include <stdio.h>
using namespace std;
int main(){
    int A,Temp;
    cin >> A;
    if(A>=100 && A<=20000 && A%100==0){
        if(A/1000!=0){
            printf("1000 = %d\n",A/1000);
        }if(A%1000/500!=0){
            printf("500 = %d\n",A%1000/500);
        }
        if(A%1000%500/100!=0){
            printf("100 = %d\n",A%1000%500/100);
        }
    }else{
        cout << "ERROR";
    }
}

File: Python/A1/052.py
n = int(input())
money = [1000,500,100]
if(100<=n<=20000 and n%100==0):
    for i in money:
        if(int(n/i)!=0):
            print(f"{i} = {int(n/i)}")
            n%=i
else:print("ERROR") 